src/backend/joanie/core/factories.py

```diff
@@ -683,6 +687,22 @@ def main_invoice(self, create, extracted, **kwargs):
 
         return None
 
+    @factory.post_generation
+    # pylint: disable=method-hidden
+    def payment_schedule(self, create, extracted, **kwargs):
+        """
+        Cast input strings for the fields `amount` and `due_date` into the appropriate types
+        """
+        if create and extracted:
+            for item in extracted:
+                if isinstance(item["due_date"], str):
+                    item["due_date"] = convert_date_str_to_date_object(item["due_date"])
+                if isinstance(item["amount"], str):
+                    item["amount"] = convert_amount_str_to_money_object(item["amount"])
+            self.payment_schedule = extracted
+            return extracted
+        return None
```

Comment:
Do we really have to do that ? Instead we could assume to use only datetime right now.

Reply:
We need to do that so in factories we are free to pass a string, and it will cast the value of `amount` and `due_date` to their types.